sktime/annotation/hidalgo.py

```diff
@@ -0,0 +1,699 @@
+    return sum(
+        [
+            binom(N1 - 1, q1)
+            * binom(N - N1, q - q1)
+            * zeta ** (q1)
+            * (1 - zeta) ** (q - q1)
+            for q1 in range(q + 1)
+        ]
+    )
+
+
+class Hidalgo(BaseSeriesAnnotator):
+    """Class to fit parameters of the HidAlgo intrinsic dimension model.
+
+    explain, reference
+
+    Parameters
+    ----------
+    metric : str, or callable, optional, default="euclidean"
+        directly passed to sklearn KNearestNeighbors,
+        must be str or callable that can be passed to KNearestNeighbors
+        distance used in the nearest neighbors part of the algorithm
+    K : int, optional, default=2
+        number of manifolds used in algorithm
+    zeta : float, optional, defualt=0.8
+        "local homogeneity level" used in the algorithm, see equation ?
+    q : int, optional, default=3
+        number of points for local Z interaction, "local homogeneity range"
+        see equation ?
+    Niter : int, optional, default=10000
+        number of Gibbs sampling iterations
+    Nreplicas : int, optional, default=1
+        number of random starts to run Gibbs sampling
+    burn_in : float, optional, default=0.9
+        percentage of Gibbs sampling iterations discarded, "burn-in fraction"
+    fixed_Z : bool, optional, default=False
+        estimate parameters with fixed z (joint posterior approximation via Gibbs)
+        z = (z_1, ..., z_K) is a latent variable introduced, where z_i = k
+        indicates point i belongs to manifold K
+    use_Potts : bool, optional, default=True
+        if using local interaction between z, see equation (4)
+    estimate_zeta : bool, optional, default=False
+        update zeta in the sampling
+    sampling_rate: int, optional, default=10
+        rate at which to save samples for each Niter
+    a : np.ArrayLike, optional, default=1.0
+        prior parameters of d
+    b : np.ArrayLike, optional, default=1.0
+        prior parameters of d
+    c : np.ArrayLike, optional, default=1.0
+        prior parameters of p
+    f : np.ArrayLike, optional, default=1.0
+        parameters of zeta
+    seed : int, optional, default = 1
+        generate random numbers with seed
+
+    Notes
+    -----
+    As described in
+    @article{allegra2020data,
+    title={Data segmentation based on the local intrinsic dimension},
+    author={Allegra, Michele and Facco, Elena and Denti, Francesco and Laio,
+            Alessandro and Mira, Antonietta},
+    journal={Scientific reports},
+    volume={10},
+    number={1},
+    pages={1--12},
+    year={2020},
+    publisher={Nature Publishing Group}
+    }
+
+    Examples
+    --------
+    >>> from sktime.annotation.hidalgo import Hidalgo
+    >>> import numpy as np
+    >>> np.random.seed(123)
+    >>> X = np.random.rand(10,3)
+    >>> X[6:, 1:] = 0
+    >>> model = Hidalgo(K=2, Niter=50, seed=10)
+    >>> fitted_model = model._fit(X)
+    >>> Z = fitted_model._predict(X)
```

Comment:
Yeah, I agree and was confused by this. In BaseSeriesAnnotator:
```
    def fit(self, X, Y=None):
        """Fit to training data.

        Parameters
        ----------
        X : pd.DataFrame
            Training data to fit model to (time series).
        Y : pd.Series, optional
            Ground truth annotations for training if annotator is supervised.

        Returns
        -------
        self :
            Reference to self.

        Notes
        -----
        Creates fitted model that updates attributes ending in "_". Sets
        _is_fitted flag to True.
        """
```

So I changed it from `y` to `Y` to avoid a unit test failure I was getting. It's not calling in any `y` now, just as a temporary fix until we redesign this BaseClass